graphql_compiler/tests/test_ir_generation_errors.py

```diff
@@ -1050,3 +1052,175 @@ def test_invalid_edge_degree_queries(self):
         for invalid_graphql in invalid_queries:
             with self.assertRaises(GraphQLCompilationError):
                 graphql_to_ir(self.schema, invalid_graphql)
+
+    def test_missing_directives_in_schema(self):
+        """Ensure that validators properly identifiy missing directives in the schema.
+
+           The schema should all directives which are supported by the graphql compiler,
+           whether they are not used in the query. Hence we raise an error when the following
```

Comment:
`whether they are not used in the query`? Maybe `even if they might not be used in the query`?